cl/tests: created a base for cluster linking tests

mmaslankaprv opened the pull request:
Create a base test class for cluster linking ducktape tests. For the cluster linking tests it should be enough to base the test of `ShadowLinkTestBase` to have the initial setup ready.

Refactored existing tests to use the new base class

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--33171e17-967a-4829-a535-76cf55f9fd27-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71494">build#71494</a></summary>


| test_class                  | test_method                             | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                |
|:----------------------------|:----------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| IdempotencyWriteCachingTest | test_idempotent_producers_write_caching | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71494#0198ef9b-bdda-43d5-97ec-82e3e56f7913 | FLAKY         | 20/21    | upstream reliability is '99.70149253731343'. current run reliability is '95.23809523809523'. drift is 4.4634 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

mmaslankaprv committed:
cl/tests: created a base for cluster linking tests

Created a base test class for cluster linking ducktape tests. For the
cluster linking tests it should be enough to base the test of
`ShadowLinkTestBase` to have the initial setup ready.

Refactored existing tests to use the new base class

Signed-off-by: Michał Maślanka <michal@redpanda.com>
